src/PowerShellEditorServices/Language/LanguageService.cs

```diff
@@ -294,72 +296,83 @@ public FindOccurrencesResult FindSymbolsInFile(ScriptFile scriptFile)
-                foreach (ScriptFile file in referencedFiles)
-                {
-                    fileMap.Add(file.FilePath, file);
-                }
+            // Make sure aliases have been loaded
+            await GetAliases();
 
-                var allFiles = workspace.EnumeratePSFiles();
-                foreach (var file in allFiles)
+            // We want to look for references first in referenced files, hence we use ordered dictionary
+            // TODO: File system case-sensitivity is based on filesystem not OS, but OS is a much cheaper heuristic
+            var fileMap = RuntimeInformation.IsOSPlatform(OSPlatform.Linux)
+                ? new OrderedDictionary()
+                : new OrderedDictionary(StringComparer.OrdinalIgnoreCase);
+            foreach (ScriptFile file in referencedFiles)
+            {
+                fileMap.Add(file.FilePath, file);
+            }
+
+            foreach (string file in workspace.EnumeratePSFiles())
+            {
+                if (!fileMap.Contains(file))
                 {
-                    if (!fileMap.Contains(file))
+                    ScriptFile scriptFile;
+                    try
                     {
-                        fileMap.Add(file, new ScriptFile(file, null, this.powerShellContext.LocalPowerShellVersion.Version));
+                        scriptFile = workspace.GetFile(file);
                     }
+                    catch (IOException)
+                    {
+                        // If the file has ceased to exist for some reason, we just skip it
+                        continue;
+                    }
+
+                    fileMap.Add(file, scriptFile);
                 }
+            }
 
-                List<SymbolReference> symbolReferences = new List<SymbolReference>();
-                foreach (var fileName in fileMap.Keys)
-                {
-                    var file = (ScriptFile)fileMap[fileName];
-                    IEnumerable<SymbolReference> symbolReferencesinFile =
-                    AstOperations
-                        .FindReferencesOfSymbol(
-                            file.ScriptAst,
-                            foundSymbol,
-                            CmdletToAliasDictionary,
-                            AliasToCmdletDictionary)
-                        .Select(
-                            reference =>
+            var symbolReferences = new List<SymbolReference>();
+            foreach (object fileName in fileMap.Keys)
+            {
+                var file = (ScriptFile)fileMap[fileName];
+                IEnumerable<SymbolReference> symbolReferencesinFile =
+                AstOperations
+                    .FindReferencesOfSymbol(
+                        file.ScriptAst,
+                        foundSymbol,
+                        CmdletToAliasDictionary,
+                        AliasToCmdletDictionary)
+                    .Select(reference =>
```

Comment:
did you want to try to clean up the LINQ usage here? We can call that out of scope if you want.

Reply:
I'm game!

Reply:
I remember when I hit this before I went and searched if we can avoid the closure allocation by turning this into a static (or instance...) method call, but I didn't find any good information on it... Would be nice if we could retain the readable composability of LINQ

Reply:
Actually, I'll open a new PR on the LINQ refactoring here, since there are other methods that would benefit. I'll see if I can do a quick benchmark before we merge